lib/src/core/result_new/result.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:hmi_core/src/core/error/failure.dart';
+/// 
+/// Containing either value of type [V] or error of type [E]
+sealed class Result<V, E> {}
+///
+/// Containing either value of type [V] or [Failure]
+typedef ResultF<V> = Result<V, Failure>;
+/// 
+/// Successful [Result] with [value].
+final class Ok<V, E> implements Result<V, E> {
+  final V value;
```

Comment:
why value is public ?

Reply:
To use it inside `switch` operator via pattern matching. Otherwise there is no point in sealed class.